exporters/otlp/otlpmetric/internal/oconf/options.go

```diff
@@ -340,23 +338,8 @@ func WithTemporalitySelector(selector metric.TemporalitySelector) GenericOption
 }
 
 func WithAggregationSelector(selector metric.AggregationSelector) GenericOption {
-	// Deep copy and validate before using.
-	wrapped := func(ik metric.InstrumentKind) aggregation.Aggregation {
```

Comment:
Why is this no longer necessary?

Reply:
I had thought this was unnecessary because we validate the aggregation in the pipeline.

However, looking closer, it looks like we only do this for an aggregation provided via a view:

https://github.com/open-telemetry/opentelemetry-go/blob/7b9fb7ac873ef242a595d4ef6bcf7c17b1512f9b/sdk/metric/view.go#L97-L105

The reader aggregation validation does not seem to occur:

https://github.com/open-telemetry/opentelemetry-go/blob/7b9fb7ac873ef242a595d4ef6bcf7c17b1512f9b/sdk/metric/pipeline.go#L313

I'll update that part of the codebase to centralize the validation of the reader returned aggregation like I had originally thought it was doing.

Reply:
Updated: ccc016bd01e8151ce608714b65c7270d5afee68f

PTAL